### PR TITLE
Fix/Do not send chunk twice on request forwarding

### LINKED ISSUE #2047: Objects forwarding does not track success progress
## Context

If a node is forwarding a request to another node, it does not track the success of the operation. It leads to a data duplication from the original caller's point of view, e.g. double header sendings:

```bash
▶ neofs-cli -r s02.neofs.devenv:8080 --bearer tmp/b.signed.json -w ./services/s3_gate/wallet.json object get --cid Gq4Zcj5tsLRhwi9RFLYu5F8W4KRzgsS6Y3mEHukk3Ecv --oid Habd23jp299wWWC5o1LF5S485cUUQfazgqsF5MMRDvTx --file trash.file
Enter password > 
 2000000 / 12000000 [====================================>------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------]  16.67% 0s
rpc error: copy payload: unexpected message instead of chunk part: *object.GetObjectPartInit

```

First time was found in #2045.

### Reproducing:

1. Make a big object (huge enough to have num of children >> num of nodes to make the bug more reproducible) to be placed on more than 1 node;
2. Send a request to a node that does not have a splitID of the object (e.g. non-container node) to force it to forward the request;
3. Make any other node that contains the part of the object not finish their response (acl problem, remove the obj physically, mb just stop it);
4. See how the assembling is starting from scratch and the requested node is answering with the data you already received.

### PR DESCRIPTION
**Pull request by carpawell**
Closes #2047.

### CODE REVIEW

**Comment by sami-nspcc** [comment]
I have recieved a notification of a new pull request. I am starting the build of images and binaries for further testing. Build number is 3756

**Comment by sami-nspcc** [comment]
I am running integration tests

**Comment by sami-nspcc** [comment]
I have recieved a notification of a new pull request. I am starting the build of images and binaries for further testing. Build number is 3757

**Comment by sami-nspcc** [comment]
I am running integration tests

**Comment by sami-nspcc** [comment]
Test run is finished. Please download the tarball from <a href="https://http.t5.fs.neo.org/get/FszwwAx3VznDysQGxBUMRj1zj3zp8VcohorVDnunrviV/HHkJd2KoTQ1oqKJVq28aVxDRZ7BNwte8e5WKxq62Mxhw">link</a>. Untar and use `allure open` to watch the report

**Comment by sami-nspcc** [comment]
I have recieved a notification of a new pull request. I am starting the build of images and binaries for further testing. Build number is 3759

**Comment by sami-nspcc** [comment]
I am running integration tests

File: pkg/services/object/get/v2/util.go [hunk 1]
```diff
@@ -646,3 +677,18 @@ func checkStatus(stV2 *status.Status) error {
 
 	return nil
 }
+
+func chunkToSend(global, local int, chunk []byte) []byte {
+	if global == 0 || global == local {
```
Comment (fyrchik) [inline pkg/services/object/get/v2/util.go:682]: What's `global == 0` condition for?

**Review by fyrchik — APPROVED** [review]

File: pkg/services/object/get/v2/util.go (hunk 1 above)
Comment (carpawell) [reply pkg/services/object/get/v2/util.go]: if `global == 0` then `local == 0` too, agree. it was the first line i wrote but then i decided that it makes the func more explicit

dropped

**Comment by sami-nspcc** [comment]
Test run is finished. Please download the tarball from <a href="https://http.t5.fs.neo.org/get/FszwwAx3VznDysQGxBUMRj1zj3zp8VcohorVDnunrviV/52SznSJeXiRBMcaBS9w3mybX8oGkZjwMRb4NcsY5EQXn">link</a>. Untar and use `allure open` to watch the report

**Comment by sami-nspcc** [comment]
I am running integration tests

**Review by fyrchik — APPROVED** [review]

**Comment by sami-nspcc** [comment]
Test run is finished. Please download the tarball from <a href="https://http.t5.fs.neo.org/get/FszwwAx3VznDysQGxBUMRj1zj3zp8VcohorVDnunrviV/7ZbnijfpDTFBGgPTQdwpEdxr2MSBrFWizA6ihF18939h">link</a>. Untar and use `allure open` to watch the report

**Pull request merged**